src/core/reducers/errors.js

```diff
@@ -0,0 +1,54 @@
+import { SET_ERROR } from 'core/constants';
+import log from 'core/logger';
+import { gettext } from 'core/utils';
+
+/*
+ * This inspects an error object and returns an array of messages from it.
+ *
+ * If the error has an API response then messages will be extracted
+ * from it. Otherwise, an array containing a generic error message is returned.
+ */
+function getMessagesFromError(error) {
+  let messages = [gettext('An unexpected error occurred')];
+  log.info('Extracting messages from error object:', error);
+
+  if (error && error.response && error.response.data) {
+    const apiMessages = [];
+
+    Object.keys(error.response.data).forEach((key) => {
```

Comment:
It would be nice to use lodash in other places where we map/iterate over objects. The `Objects.keys(obj).map((k) => obj[k])` thing is a bit verbose and `_.map(obj, (v, k) => v)` is definitely nicer. We could look at doing that later though.